modules/currencies/src/lib.rs

```diff
@@ -0,0 +1,770 @@
+		if currency_id.is_erc20() {
+			Default::default()
+		} else if currency_id == T::GetNativeCurrencyId::get() {
+			T::NativeCurrency::minimum_balance()
+		} else {
+			T::MultiCurrency::minimum_balance(currency_id)
+		}
+	}
+
+	fn total_issuance(currency_id: Self::CurrencyId) -> Self::Balance {
+		if let CurrencyId::ERC20(contract) = currency_id {
+			T::EVMBridge::total_supply(InvokeContext {
+				contract,
+				source: Default::default(),
+			})
+			.unwrap_or_default()
+		} else if currency_id == T::GetNativeCurrencyId::get() {
+			T::NativeCurrency::total_issuance()
+		} else {
+			T::MultiCurrency::total_issuance(currency_id)
+		}
+	}
+
+	fn total_balance(currency_id: Self::CurrencyId, who: &T::AccountId) -> Self::Balance {
+		if let CurrencyId::ERC20(contract) = currency_id {
+			if let Some(address) = T::AddressMapping::to_evm_address(&who) {
+				let context = InvokeContext {
+					contract,
+					source: Default::default(),
+				};
+				return T::EVMBridge::balance_of(context, address).unwrap_or_default();
+			}
+			Default::default()
+		} else if currency_id == T::GetNativeCurrencyId::get() {
+			T::NativeCurrency::total_balance(who)
+		} else {
+			T::MultiCurrency::total_balance(currency_id, who)
+		}
+	}
+
+	fn free_balance(currency_id: Self::CurrencyId, who: &T::AccountId) -> Self::Balance {
+		if let CurrencyId::ERC20(contract) = currency_id {
+			if let Some(address) = T::AddressMapping::to_evm_address(&who) {
+				let context = InvokeContext {
+					contract,
+					source: Default::default(),
+				};
+				return T::EVMBridge::balance_of(context, address).unwrap_or_default();
+			}
+			Default::default()
+		} else if currency_id == T::GetNativeCurrencyId::get() {
+			T::NativeCurrency::free_balance(who)
+		} else {
+			T::MultiCurrency::free_balance(currency_id, who)
+		}
+	}
+
+	fn ensure_can_withdraw(currency_id: Self::CurrencyId, who: &T::AccountId, amount: Self::Balance) -> DispatchResult {
+		if let CurrencyId::ERC20(contract) = currency_id {
+			let address = T::AddressMapping::to_evm_address(&who).ok_or(Error::<T>::AccountNotFound)?;
+			let balance = T::EVMBridge::balance_of(
+				InvokeContext {
+					contract,
+					source: Default::default(),
+				},
+				address,
+			)
+			.unwrap_or_default();
+			ensure!(balance >= amount, Error::<T>::BalanceTooLow);
+			Ok(())
+		} else if currency_id == T::GetNativeCurrencyId::get() {
+			T::NativeCurrency::ensure_can_withdraw(who, amount)
+		} else {
+			T::MultiCurrency::ensure_can_withdraw(currency_id, who, amount)
+		}
+	}
+
+	fn transfer(
+		currency_id: Self::CurrencyId,
+		from: &T::AccountId,
+		to: &T::AccountId,
+		amount: Self::Balance,
+	) -> DispatchResult {
+		if amount.is_zero() || from == to {
+			return Ok(());
+		}
+
+		if let CurrencyId::ERC20(contract) = currency_id {
+			let source = T::AddressMapping::to_evm_address(&from).ok_or(Error::<T>::AccountNotFound)?;
+			let address = T::AddressMapping::to_evm_address(&to).ok_or(Error::<T>::AccountNotFound)?;
```

Comment:
this means we can't have treasury or other system account to hold ERC20 tokens... let me think how can we improve this...

Reply:
https://github.com/AcalaNetwork/Acala/pull/635 should help this